packages/arb-token-bridge-ui/src/pages/api/withdrawals.ts

```diff
@@ -0,0 +1,138 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { gql } from '@apollo/client'
+import {
+  FetchWithdrawalsFromSubgraphResult,
+  getL2SubgraphClient
+} from 'token-bridge-sdk'
+
+// Extending the standard NextJs request with Withdrawal-params
+type NextApiRequestWithWithdrawalParams = NextApiRequest & {
+  query: {
+    l2ChainId: string
+    address: string
+    search?: string
+    page?: string
+    pageSize?: string
+    fromBlock?: string
+    toBlock?: string
+  }
+}
+
+type WithdrawalResponse = {
+  data: FetchWithdrawalsFromSubgraphResult[]
+  message?: string // in case of any error
+}
+
+export default async function handler(
+  req: NextApiRequestWithWithdrawalParams,
+  res: NextApiResponse<WithdrawalResponse>
+) {
+  try {
+    const {
+      address,
+      search = '',
+      l2ChainId,
+      page = '0',
+      pageSize = '10',
+      fromBlock,
+      toBlock
+    } = req.query
+
+    // validate method
+    if (req.method !== 'GET') {
+      res
+        .status(400)
+        .send({ message: `invalid_method: ${req.method}`, data: [] })
+      return
+    }
+
+    // validate the request parameters
+    if (!l2ChainId || !address) {
```

Comment:
nit: most likely not big deal now, but could it be that we'd want to add more checks in the future? This way we could remove the need for another `if` chain check on top.

```suggestion
 const errorMessages = []
  if (!l2ChainId) errorMessages.push('<l2ChainId> is required')
  if (!address) errorMessages.push('<address> is required')

  if (errorMessages.length) {
    res.status(400).json({
      message: `incomplete request: ${errorMessage.join(', ')}`,
      data: []
    })
  }
```

Reply:
Nice. Much cleaner. Done.